### ISSUE TITLE
[BUG?] Message routing using multiple external component strange behaviour.

### ISSUE THREAD

**Issue by Kyos** [processone/ejabberd#2278]
> What version of ejabberd are you using?

18.01

> What operating system (version) are you using?

Debian stretch (x86_64)

> How did you install ejabberd (source, package, distribution)?

package (deb)

> What did not work as expected? Are there error messages in the log? What
> was the unexpected behaviour? What was the expected result?

I have notice a strange behaviour using multiple external components in ejabberd. Here is the configuration we use : 

  -
    port: 5288
    module: ejabberd_service
    hosts:
      "component.test.com":
        password: "XXXXX"
      "component2.test.com":
        password: "XXXXX"

When we have the 2 components connected (one on component.test.com & another one on component2.test.com), the message broadcasting is acting very weird. If I send a message to component2.test.com, component.test.com is sometime receiving it.

I was wondering if it was the expected behaviour or a bug ?

**Comment by zinid** [comment]
Just connect them on different ports.

**Comment by Kyos** [comment]
Thank you for the fast answer ! 

I agree that this could work, but since we plan to have to have around 40 components this might be a small madness to maintain.

I don't think this is XEP compliant since we bind a component on a vhost it is not supposed to be bound. 

https://github.com/processone/ejabberd/blob/master/src/ejabberd_service.erl#L165

Shouldn't we just check that the vhost is in the dict instead of looping ? 

Thanks by advance :)

**Comment by zinid** [comment]
Well, this is some legacy behaviour which I don't remember why was implemented this way.
Right now, you can work around this if it's ok for you to have a single password for all components, in this case you can configure it as simple as:
```yaml
listen:
  -
    port: 5288
    module: ejabberd_service
    password: "XXXXX"
...
```

**Comment by zinid** [comment]
Fixed in 032f79629288a30aa0288f77ec06d210de943d0c

**Comment by Kyos** [comment]
Thank you very much :)
